smartsim/_core/mli/workermanager.py

```diff
@@ -0,0 +1,976 @@
+        :param transform_result: The result of transforming inputs for model consumption
+        :return: The result of inference wrapped in an ExecuteResult"""
+
+    @staticmethod
+    @abstractmethod
+    def transform_output(
+        request: InferenceRequest,
+        execute_result: ExecuteResult,
+    ) -> OutputTransformResult:
+        """Given inference results, perform transformations required to
+        transmit results to the requestor.
+        :param request: The request that triggered the pipeline
+        :param execute_result: The result of inference wrapped in an ExecuteResult
+        :return:"""
+
+    @staticmethod
+    @abstractmethod
+    def serialize_reply(reply: InferenceReply) -> bytes:
+        """Given an output, serialize to bytes for transport
+        :param reply: The result of the inference pipeline
+        :return: a byte-serialized version of the reply"""
+
+
+class SampleTorchWorker(MachineLearningWorkerBase):
+    """A minimum implementation of a worker that executes a PyTorch model"""
+
+    @staticmethod
+    def deserialize(data_blob: bytes) -> InferenceRequest:
+        request: InferenceRequest = pickle.loads(data_blob)
+        return request
+
+    @staticmethod
+    def load_model(
+        request: InferenceRequest, fetch_result: FetchModelResult
+    ) -> ModelLoadResult:
+        model_bytes = fetch_result.model_bytes or request.raw_model
+        if not model_bytes:
+            raise ValueError("Unable to load model without reference object")
+
+        model: torch.nn.Module = torch.load(io.BytesIO(model_bytes))
+        result = ModelLoadResult(model)
+        return result
+
+    @staticmethod
+    def transform_input(
+        request: InferenceRequest, fetch_result: InputFetchResult
+    ) -> InputTransformResult:
+        result = [torch.load(io.BytesIO(item)) for item in fetch_result.inputs]
+        return InputTransformResult(result)
+        # return data # note: this fails copy test!
+
+    @staticmethod
+    def execute(
+        request: InferenceRequest,
+        load_result: ModelLoadResult,
+        transform_result: InputTransformResult,
+    ) -> ExecuteResult:
+        """Execute an ML model on the given inputs"""
+        if not load_result.model:
+            raise sse.SmartSimError("Model must be loaded to execute")
+
+        model = load_result.model
+        results = [model(tensor) for tensor in transform_result.transformed]
+
+        execute_result = ExecuteResult(results)
+        return execute_result
+
+    @staticmethod
+    def transform_output(
+        request: InferenceRequest,
+        execute_result: ExecuteResult,
+    ) -> OutputTransformResult:
+        transformed = [item.clone() for item in execute_result.predictions]
+        return OutputTransformResult(transformed)
+
+    @staticmethod
+    def serialize_reply(reply: InferenceReply) -> bytes:
+        return pickle.dumps(reply)
+
+
+class IntegratedTorchWorker(MachineLearningWorkerBase):
```

Comment:
Move this to the test directory in a consistent way as the other test classes.

Reply:
moved all mock/test implementations into `tests/mli/*.py`